Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -60,20 +60,29 @@ bool VirtualFileSystem::mount_point_exists_at_inode(InodeIdentifier inode_identi
     });
 }
 
+void VirtualFileSystem::mount_and_increment_filesystem_mounted_count(NonnullOwnPtr<Mount> mount)
+{
+    m_mounts.with([&](auto& mounts) {
+        mount->guest_fs().increment_mounted_count({});
+        mounts.append(move(mount));
+    });
+}
+
 ErrorOr<void> VirtualFileSystem::mount(FileSystem& fs, Custody& mount_point, int flags)
 {
     auto new_mount = TRY(adopt_nonnull_own_or_enomem(new (nothrow) Mount(fs, &mount_point, flags)));
-    return m_mounts.with([&](auto& mounts) -> ErrorOr<void> {
+    return m_mounts.with([&](auto&) -> ErrorOr<void> {
         auto& inode = mount_point.inode();
-        dbgln("VirtualFileSystem: Mounting {} at inode {} with flags {}",
+        dbgln("VirtualFileSystem: FileSystemID {}, Mounting {} at inode {} with flags {}",
+            fs.fsid(),
             fs.class_name(),
             inode.identifier(),
             flags);
         if (mount_point_exists_at_inode(inode.identifier())) {
             dbgln("VirtualFileSystem: Mounting unsuccessful - inode {} is already a mount-point.", inode.identifier());
             return EBUSY;
         }
-        mounts.append(move(new_mount));
+        mount_and_increment_filesystem_mounted_count(move(new_mount));
```

Comment:
this will lock m_mounts again in the `mount_and_increment_filesystem_mounted_count` method. 

Do we really want m_mounts to be recursively locked?

This is only called in two places, would it be better to have an internal "bookeep_new_mount" method for that part that requires the spinlock to be held?